File: wk3/intersections.py
# intersections.py

from newton import solve

def p4(debug_solve=False):
	"""
	Problem 4 from Week 3 HW, find intersects of two functions.
	"""
	from numpy import sqrt
	for x0 in [-2.2, -1.6, .75, 1.4]:
		print " "  # blank line
		# print "for init of the funcial guess %22.15e" % x0
		x,iters = solve(fvals_inter, x0, debug=debug_solve)
		print "solve returns x = %22.15e after %i iterations " % (x,iters)
		fx,fpx = fvals_inter(x)
		print "the value of f(x) is %22.15e" % fx

def fvals_inter(x):
	import math
	pi = math.pi
	f = 0.6*x**2 + x*math.cos(pi*x) - 1
	fp = -pi*x*math.sin(pi*x) + 1.2*x + math.cos(pi*x)

	return f, fp

File: wk3/fortran-p7/intersections.f90
! $UWHPSC/wk3/fortran/intersections.f90

program intersections

    use newton, only: solve
    use functions, only: f, fp, g_2

    implicit none
    real(kind=8) :: x, x0, fx
    real(kind=8) :: x0vals(4)
    integer :: iters, itest
	logical :: debug         ! set to .true. or .false.

    print *, "Test routine for intersections of g_1 and g_2"
    debug = .true.

    ! values to test as x0:
    x0vals = (/-2.2d0, -1.6d0, -0.75d0, 1.4d0/)

    do itest=1,4
        x0 = x0vals(itest)
		print *, ' '  ! blank line
        call solve(f, fp, x0, x, iters, debug)

        print 11, x, iters
11      format('solver returns x = ', es22.15, ' after', i3, ' iterations')

        fx = g_2(x)
        print 12, fx
12      format('the value of f(x) is ', es22.15)

        enddo

end program intersections
